File: 카카오 겨울인턴2019/크레인 인형뽑기 게임.js
function solution(board, moves) {
    let answer = 0; // 카운트
    const BLANK = 0;
    let length = board[0].length
    let stack = []

    for (const move of moves) { // 1,5,3,5 ...
        for (let i = 0; i < length; i++) { // 위에서 부터 탐색
            if (board[i][move - 1] !== BLANK) { // 인형이 있다면
                let doll = board[i][move - 1]
                board[i][move - 1] = BLANK // 인형을 뽑았으니 빈칸으로
                if (stack && stack[stack.length - 1] === doll) {
                    stack.pop()
                    answer += 2
                } else { // 인형이 다르거나 비었을 때
                    stack.push(doll)
                }
                break; // 빠져나가기
            }
        }
    }
    return answer;
}

File: 카카오 겨울인턴2019/크레인 인형뽑기 게임.py
def solution(board, moves):
    BLANK, answer = 0, 0
    stack = []
    new_board = [list(line[::-1]) for line in zip(*board)] # 행-열 바꾸기
    for i, line in enumerate(new_board): # 0 제거
        new_board[i] = [doll for doll in line if doll != BLANK]
    
    for move in moves:
        if new_board[move-1]: # 인형이 있으면
            doll = new_board[move-1].pop() # 꺼내기
            if stack and stack[-1] == doll: # 인형이 같을 떄
                answer += 2
                stack.pop()
            else: # 다를때
                stack.append(doll)
            
    return answer
